### PR TITLE
Reporting: Render only selected fields in MessageList

### PR DESCRIPTION
**Pull request by janheise**
## Description

This PR adds the necessary changes in the Server package needed, to support rendering only the selected fields in a MessageTable in reporting formats other than PDF.

see https://github.com/Graylog2/graylog-plugin-enterprise/pull/12368 for more details

/jpd https://github.com/Graylog2/graylog-plugin-enterprise/pull/12368

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have requested a documentation update.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/test/java/org/graylog/plugins/views/search/rest/export/response/ExportTabularResultResponseTest.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ void testCreationFromMessageList() throws Exception {
         final ObjectMapper objectMapper = new ObjectMapperProvider().get();
         final byte[] bytes = Resources.toByteArray(Resources.getResource("org/graylog/plugins/views/search/rest/export/response/sample_message_list_result.json"));
         final MessageList.Result messageListResult = objectMapper.readValue(bytes, MessageList.Result.class);
-        final ExportTabularResultResponse response = ExportTabularResultResponse.fromMessageListResult(messageListResult);
+        final ExportTabularResultResponse response = ExportTabularResultResponse.fromMessageListResult(List.of(), messageListResult);
```
Comment (luk-kaminski) [inline graylog2-server/src/test/java/org/graylog/plugins/views/search/rest/export/response/ExportTabularResultResponseTest.java:84]: I guess that you should also add a unit test for a scenario when list of fields is not empty.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java [hunk 2]
```diff
@@ -77,14 +77,13 @@ public static ExportTabularResultResponse fromPivotResult(final PivotResult pivo
...
+    public static ExportTabularResultResponse fromMessageListResult(final List<String> fields, final MessageList.Result m) {
         if(m.messages().isEmpty()) {
             return new ExportTabularResultResponse(List.of(), List.of());
         }
 
-        final var first = m.messages().get(0);
-
-        final var header = first.message().keySet().stream().toList();
+        // iterating over all messages for the header to make sure that we collect all headers from all messages
+        final var header = (fields != null && !fields.isEmpty()) ? fields : m.messages().stream().flatMap(s -> s.message().keySet().stream()).collect(Collectors.toSet()).stream().toList();
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java]: `...collect(Collectors.toSet()).stream().toList()`

What about collecting to List and using "distinct()" instead of creating two separate collections (Set and List) ?

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java [hunk 3]
```diff
@@ -77,14 +77,13 @@ public static ExportTabularResultResponse fromPivotResult(final PivotResult pivo
         return new ExportTabularResultResponse(header, dataRows);
     }
 
-    public static ExportTabularResultResponse fromMessageListResult(final MessageList.Result m) {
+    public static ExportTabularResultResponse fromMessageListResult(final List<String> fields, final MessageList.Result m) {
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java:80]: `final List<String> fields` should be `final Set<String> fields`, I guess...

Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java:80]: or is the ordering important?

Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java:80]: you can order the columns in the FE, so I think we should keep it

File: graylog2-server/src/test/java/org/graylog/plugins/views/search/rest/export/response/ExportTabularResultResponseTest.java (hunk 1 above)
Comment (janheise) [reply graylog2-server/src/test/java/org/graylog/plugins/views/search/rest/export/response/ExportTabularResultResponseTest.java:84]: added it
